### PR TITLE
Make InterestGroup.ads optional

### PR DESCRIPTION
**Pull request by taymonbeal**
This matches the explainer and will be useful once more properties are added.

### CODE REVIEW

**Review by jeffkaufman — APPROVED** [review]

File: lib/public_api.ts [hunk 1]
```diff
@@ -184,8 +184,16 @@ export class FledgeShim {
...
   /**
-   * Creates a new registration in this browser for a specified interest group
-   * and stores it client-side.
+   * Creates or updates a registration in this browser for a specified interest
+   * group and stores it client-side.
+   *
+   * If `group.name` is the name of an existing interest group, each of that
+   * existing interest group's properties is overwritten with the corresponding
+   * property from `group`, unless that property from `group` is absent or
+   * `undefined`. For each property of `group` that is absent or undefined, the
```
[the remark this answers is not in the document]
Comment (jeffkaufman) [reply lib/public_api.ts:193]: I wonder if maybe we are misreading the spec?

File: frame/handler_test.ts [hunk 2]
```diff
@@ -82,6 +82,19 @@ describe("handleRequest", () => {
     expect([...(await getAllAds())]).toEqual(ads);
   });
 
+  it("should do nothing when joining an interest group with no ads", async () => {
```
[the remark this answers is not in the document]
Comment (taymonbeal) [reply frame/handler_test.ts:85]: Yeah, this is temporary, I've already got a commit on disk that replaces it with something else.

File: lib/public_api.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (taymonbeal) [reply lib/public_api.ts:193]: Filing a bug SGTM. Will stick with current implementation pending feedback from proposal owners.

**Review by jeffkaufman — APPROVED** [review]

File: lib/public_api.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (jeffkaufman) [reply lib/public_api.ts:193]: Filed https://github.com/WICG/turtledove/issues/188

**Pull request merged**